src/mono/mono/mini/aot-runtime.c

```diff
@@ -4920,14 +4920,35 @@ mono_aot_get_method (MonoMethod *method, MonoError *error)
 	MonoClass *klass = method->klass;
 	MonoMethod *orig_method = method;
 	guint32 method_index;
-	MonoAotModule *amodule = m_class_get_image (klass)->aot_module;
+	MonoImage *image = m_class_get_image (klass);
 	guint8 *code = NULL;
 	gboolean cache_result = FALSE;
 	ERROR_DECL (inner_error);
 
 	error_init (error);
 
-	if (!amodule)
+	if (!(image->aot_module)) {
+		// aot_module was uninitialized
+		MonoMethodHeader *header = mono_method_get_header_checked (method, error);
+		if (!header) {
+			return NULL;
+		} else if (header->code_size != 0) {
+			return NULL;
```

Comment:
this needs to be something like `header->code_size != 0  && !(fullaot && !interpreter)`.
If we're in full AOT without an interpreter we have no choice but to wait for the AOT module to get loaded - there's no other fallback

Reply:
> It would go through the next branch after this.

oh, because with fullaot we always strip the IL so `code_size` will be 0?

> We wait until aot_module is not NULL any more.

Let's not do that.  That will deadlock.  Actually, can you add a comment next to the loop to remind us not to do that in the future.

Reply:
iOS app always has IL stripped out. I would expect `code_size` to be 0.

Will add a comment.